Dockerfile

```diff
@@ -1,56 +1,129 @@
-FROM trailofbits/polytracker-llvm:bbdb72d136af011db9710a4d44a79e6d5059cc1a
+FROM ubuntu:jammy as base
 
 LABEL org.opencontainers.image.authors="evan.sultanik@trailofbits.com"
-ENV DEBIAN_FRONTEND=noninteractive
-RUN apt-get -y update  && \
-      apt-get install -y \
-            ninja-build                                     \
-            python3-pip                                     \
-            python3.8-dev                                   \
-            libgraphviz-dev                                 \
-            libjpeg-dev                                     \
-            graphviz                                        \
-            vim                                             \
-            file                                            \
-            gdb                                             \
-            libncurses5-dev                                 \
-            apt-transport-https                             \
-            ca-certificates                                 \
-            libstdc++-10-dev
+
+ARG BUILD_TYPE="Release"
+ARG PARALLEL_LINK_JOBS=1
+ARG LLVM_TARGET_NAME=X86
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y update
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install \
+  ninja-build                                         \
+  python3-pip                                         \
+  python3.8-dev                                       \
+  python-is-python3                                   \
+  ca-certificates                                     \
+  libstdc++-10-dev                                    \
+  golang                                              \
+  clang-12                                            \
+  cmake                                               \
+  git                                                 \
+  file
+
+RUN update-alternatives --install /usr/bin/opt opt /usr/bin/opt-12 10
+RUN update-alternatives --install /usr/bin/llvm-link llvm-link /usr/bin/llvm-link-12 10
+RUN update-alternatives --install /usr/bin/llvm-ar llvm-ar /usr/bin/llvm-ar-12 10
+RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 10
+RUN update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-12 10
```

Comment:
>Does this mean we are using llvm-12? Aren't we using llvm-13 in the current branch?

We're installing `clang` and related bitcode utilities via `apt` in version 12 because the `clang-13` package contains version `13.0.1` which has weird behavior wrt `-Werror`. The flag seems to be raised even if the user doesn't explicitly specify so. I recall seeing some discussion about this in official LLVM channels and it was done to mimic `gcc` behavior. `MuPDF` doesn't build with `clang-13` installed from `apt` for example. The version we use in `polytracker-llvm` is `13.0.0`.

> It seems from the clone below that we build libcxx from version 13. How does this work?

The version mismatch here doesn't really matter in my opinion. We're building `libcxx` using `clang-12`.

Reply:
I agree that the version mismatch doesn't really matter in this case. It would be nice to copy/paste this reasoning as a comment, though.
```suggestion
RUN update-alternatives --install /usr/bin/opt opt /usr/bin/opt-12 10
RUN update-alternatives --install /usr/bin/llvm-link llvm-link /usr/bin/llvm-link-12 10
RUN update-alternatives --install /usr/bin/llvm-ar llvm-ar /usr/bin/llvm-ar-12 10
RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 10
RUN update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-12 10
# We install clang and related bitcode utilities via apt in version 12 because the clang-13 package contains version 13.0.1 which has weird behavior wrt -Werror.
# The flag seems to be raised even if the user doesn't explicitly specify so.
# We believe this is intentional on the part of LLVM to mimic gcc behavior.
# MuPDF doesn't build with clang-13 installed from apt, for example.
```

Reply:
Added a comment!